src/Android/Xamarin.Android/Samples/MapView/ChooseCameraController/ChooseCameraController.cs

```diff
@@ -0,0 +1,178 @@
+
+            // Add the surface to the scene.
+            myScene.BaseSurface = surface;
+
+            // Create a graphics overlay for the scene.
+            GraphicsOverlay sceneGraphicsOverlay = new GraphicsOverlay()
+            {
+                SceneProperties = new LayerSceneProperties(SurfacePlacement.Absolute)
+            };
+            _mySceneView.GraphicsOverlays.Add(sceneGraphicsOverlay);
+
+            // Location at the crater.
+            MapPoint craterLocation = new MapPoint(-109.929589, 38.437304, 1700, SpatialReferences.Wgs84);
+
+            // Create the plane symbol and make it 10x larger (to be the right size relative to the scene).
+            ModelSceneSymbol planeSymbol;
+            try
+            {
+                planeSymbol = await ModelSceneSymbol.CreateAsync(_modelUri, 10.0);
+            }
+            catch (Exception ex)
+            {
+                System.Diagnostics.Debug.WriteLine(ex.Message);
+                CreateErrorDialog("Loading plane model failed. Sample failed to initialize.");
+                return;
+            }
+
+            // Create a graphic using the plane symbol.
+            Graphic planeGraphic = new Graphic(new MapPoint(craterLocation.X, craterLocation.Y, 5000.0, SpatialReferences.Wgs84), planeSymbol);
+            sceneGraphicsOverlay.Graphics.Add(planeGraphic);
+
+            // Instantiate a new camera controller which orbits a geo element.
+            _orbitPlaneCameraController = new OrbitGeoElementCameraController(planeGraphic, 300.0)
+            {
+                CameraPitchOffset = 30,
+                CameraHeadingOffset = 150
+            };
+
+            // Instantiate a new camera controller which orbits a location.
+            _orbitCraterCameraController = new OrbitLocationCameraController(craterLocation, 6000.0)
+            {
+                CameraPitchOffset = 3,
+                CameraHeadingOffset = 150
+            };
+
+            // Set the starting camera controller.
+            _mySceneView.CameraController = _orbitPlaneCameraController;
+
+            // Add the scene to the view.
+            _mySceneView.Scene = myScene;
+        }
+
+        private void ChangeCameraController(string camera)
+        {
+            switch (camera)
+            {
+                case "Orbit camera around plane":
+                    // Switch to the plane camera controller.
+                    _mySceneView.CameraController = _orbitPlaneCameraController;
+                    break;
+
+                case "Orbit camera around crater":
+                    // Switch to the crater camera controller.
+                    _mySceneView.CameraController = _orbitCraterCameraController;
+                    break;
+
+                case "Free pan around the globe":
+                    // Switch to a globe camera controller, which is free pan.
+                    _mySceneView.CameraController = new GlobeCameraController();
```

Comment:
I wonder if it would be better to create the globe camera controller once and re-use it, following the pattern used with the other camera controllers. If there is a specific reason to re-create each time, I recommend documenting that.

Reply:
There is not a specific reason to recreate each time. Creating a readonly GlobeCameraController in the private variables results in the same behavior. I will change the samples accordingly.